`singletons-base` test suite fails on Windows

If you run the `singletons-base-3.1` test suite on Windows, most of the tests will fail:

<details>

```
Test suite singletons-base-test-suite: RUNNING...
Testsuite
  Singletons
    Nat:                  FAIL (6.56s)
      1c1
      < Singletons/Nat.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Nat.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.Nat"' to rerun this test only.
    Empty:                FAIL (2.54s)
      1c1
      < Singletons/Empty.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Empty.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.Empty"' to rerun this test only.
    Maybe:                FAIL (4.88s)
      1c1
      < Singletons/Maybe.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Maybe.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Maybe/' to rerun this test only.
    BoxUnBox:             FAIL (2.56s)
      1c1
      < Singletons/BoxUnBox.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\BoxUnBox.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/BoxUnBox/' to rerun this test only.
    Operators:            SKIP

      Use -p '/Operators/' to rerun this test only.
    HigherOrder:          SKIP

      Use -p '/HigherOrder/' to rerun this test only.
    Contains:             FAIL (4.32s)
      1c1
      < Singletons/Contains.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Contains.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Contains/' to rerun this test only.
    AsPattern:            SKIP

      Use -p '/AsPattern/' to rerun this test only.
    DataValues:           SKIP

      Use -p '/DataValues/' to rerun this test only.
    EqInstances:          SKIP

      Use -p '/EqInstances/' to rerun this test only.
    CaseExpressions:      FAIL (4.37s)
      1c1
      < Singletons/CaseExpressions.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\CaseExpressions.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/CaseExpressions/' to rerun this test only.
    Star:                 SKIP

      Use -p '/Star/' to rerun this test only.
    ReturnFunc:           SKIP

      Use -p '/ReturnFunc/' to rerun this test only.
    Lambdas:              FAIL (6.33s)
      1c1
      < Singletons/Lambdas.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Lambdas.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.Lambdas"' to rerun this test only.
    LambdasComprehensive: SKIP

      Use -p '/LambdasComprehensive/' to rerun this test only.
    Error:                FAIL (3.84s)
      1c1
      < Singletons/Error.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Error.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Error/' to rerun this test only.
    TopLevelPatterns:     FAIL (4.90s)
      1c1
      < Singletons/TopLevelPatterns.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\TopLevelPatterns.hs:(0,0)-(0,0): Splicing declarations
      78c78
      < Singletons/TopLevelPatterns.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\TopLevelPatterns.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/TopLevelPatterns/' to rerun this test only.
    LetStatements:        SKIP

      Use -p '/LetStatements/' to rerun this test only.
    LambdaCase:           FAIL (4.76s)
      1c1
      < Singletons/LambdaCase.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\LambdaCase.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/LambdaCase/' to rerun this test only.
    Sections:             SKIP

      Use -p '/Sections/' to rerun this test only.
    PatternMatching:      SKIP

      Use -p '/PatternMatching/' to rerun this test only.
    Records:              FAIL (3.91s)
      1c1
      < Singletons/Records.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Records.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Records/' to rerun this test only.
    T29:                  FAIL (6.28s)
      1c1
      < Singletons/T29.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T29.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.T29"' to rerun this test only.
    T33:                  FAIL (3.77s)
      1c1
      < Singletons/T33.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T33.hs:(0,0)-(0,0): Splicing declarations
      29c29
      < Singletons/T33.hs:0:0: warning:
      ---
      > Singletons\T33.hs:0:0: warning:
      35c35
      < Singletons/T33.hs:0:0: warning:
      ---
      > Singletons\T33.hs:0:0: warning:

      Use -p '$0=="Testsuite.Singletons.T33"' to rerun this test only.
    T54:                  FAIL (3.87s)
      1c1
      < Singletons/T54.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T54.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T54/' to rerun this test only.
    Classes:              SKIP

      Use -p '$0=="Testsuite.Singletons.Classes"' to rerun this test only.
    Classes2:             SKIP

      Use -p '/Classes2/' to rerun this test only.
    FunDeps:              FAIL (3.86s)
      1c1
      < Singletons/FunDeps.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\FunDeps.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/FunDeps/' to rerun this test only.
    T78:                  FAIL (3.55s)
      1c1
      < Singletons/T78.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T78.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T78/' to rerun this test only.
    OrdDeriving:          FAIL (18.21s)
      1c1
      < Singletons/OrdDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\OrdDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/OrdDeriving/' to rerun this test only.
    BoundedDeriving:      FAIL (4.12s)
      1c1
      < Singletons/BoundedDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\BoundedDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Singletons.BoundedDeriving/' to rerun this test only.
    BadBoundedDeriving:   FAIL (1.46s)
      2c2
      < Singletons/BadBoundedDeriving.hs:0:0: error:
      ---
      > Singletons\BadBoundedDeriving.hs:0:0: error:

      Use -p '/BadBoundedDeriving/' to rerun this test only.
    EnumDeriving:         FAIL (4.18s)
      1c1
      < Singletons/EnumDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\EnumDeriving.hs:(0,0)-(0,0): Splicing declarations
      145c145
      < Singletons/EnumDeriving.hs:0:0:: Splicing declarations
      ---
      > Singletons\EnumDeriving.hs:0:0:: Splicing declarations

      Use -p '/Singletons.EnumDeriving/' to rerun this test only.
    BadEnumDeriving:      FAIL (1.59s)
      2c2
      < Singletons/BadEnumDeriving.hs:0:0: error:
      ---
      > Singletons\BadEnumDeriving.hs:0:0: error:

      Use -p '/BadEnumDeriving/' to rerun this test only.
    Fixity:               FAIL (3.57s)
      1c1
      < Singletons/Fixity.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Fixity.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Fixity/' to rerun this test only.
    Undef:                FAIL (3.20s)
      1c1
      < Singletons/Undef.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Undef.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Undef/' to rerun this test only.
    T124:                 FAIL (3.40s)
      1c1
      < Singletons/T124.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T124.hs:(0,0)-(0,0): Splicing declarations
      30c30
      < Singletons/T124.hs:0:0:: Splicing expression
      ---
      > Singletons\T124.hs:0:0:: Splicing expression

      Use -p '/T124/' to rerun this test only.
    T136:                 FAIL (5.09s)
      1c1
      < Singletons/T136.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T136.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.T136"' to rerun this test only.
    T136b:                FAIL (2.29s)
      1c1
      < Singletons/T136b.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T136b.hs:(0,0)-(0,0): Splicing declarations
      25c25
      < Singletons/T136b.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T136b.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T136b/' to rerun this test only.
    T153:                 OK (1.19s)
    T157:                 OK (1.24s)
    T159:                 FAIL (2.42s)
      1c1
      < Singletons/T159.hs:0:0:: Splicing declarations
      ---
      > Singletons\T159.hs:0:0:: Splicing declarations
      159c159
      < Singletons/T159.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T159.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T159/' to rerun this test only.
    T167:                 FAIL (4.22s)
      1c1
      < Singletons/T167.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T167.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T167/' to rerun this test only.
    T145:                 FAIL (1.99s)
      1c1
      < Singletons/T145.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T145.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T145/' to rerun this test only.
    PolyKinds:            FAIL (2.21s)
      1c1
      < Singletons/PolyKinds.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\PolyKinds.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.PolyKinds"' to rerun this test only.
    PolyKindsApp:         FAIL (1.80s)
      1c1
      < Singletons/PolyKindsApp.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\PolyKindsApp.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/PolyKindsApp/' to rerun this test only.
    T150:                 SKIP

      Use -p '/T150/' to rerun this test only.
    T160:                 FAIL (3.53s)
      1c1
      < Singletons/T160.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T160.hs:(0,0)-(0,0): Splicing declarations
      63c63
      < Singletons/T160.hs:0:0: error:
      ---
      > Singletons\T160.hs:0:0: error:

      Use -p '/T160/' to rerun this test only.
    T163:                 FAIL (2.41s)
      1c1
      < Singletons/T163.hs:0:0:: Splicing declarations
      ---
      > Singletons\T163.hs:0:0:: Splicing declarations

      Use -p '/T163/' to rerun this test only.
    T166:                 FAIL (1.95s)
      1c1
      < Singletons/T166.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T166.hs:(0,0)-(0,0): Splicing declarations
      139c139
      < Singletons/T166.hs:0:0: error:
      ---
      > Singletons\T166.hs:0:0: error:

      Use -p '/T166/' to rerun this test only.
    T172:                 FAIL (4.14s)
      1c1
      < Singletons/T172.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T172.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T172/' to rerun this test only.
    T175:                 FAIL (2.59s)
      1c1
      < Singletons/T175.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T175.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T175/' to rerun this test only.
    T176:                 FAIL (4.41s)
      1c1
      < Singletons/T176.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T176.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T176/' to rerun this test only.
    T178:                 FAIL (5.37s)
      1c1
      < Singletons/T178.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T178.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T178/' to rerun this test only.
    T183:                 FAIL (4.71s)
      1c1
      < Singletons/T183.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T183.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T183/' to rerun this test only.
    T184:                 FAIL (7.75s)
      1c1
      < Singletons/T184.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T184.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T184/' to rerun this test only.
    T187:                 FAIL (3.55s)
      1c1
      < Singletons/T187.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T187.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T187/' to rerun this test only.
    T190:                 FAIL (4.92s)
      1c1
      < Singletons/T190.hs:0:0:: Splicing declarations
      ---
      > Singletons\T190.hs:0:0:: Splicing declarations

      Use -p '/T190/' to rerun this test only.
    ShowDeriving:         FAIL (13.09s)
      1c1
      < Singletons/ShowDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\ShowDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Singletons.ShowDeriving/' to rerun this test only.
    EmptyShowDeriving:    FAIL (4.07s)
      1c1
      < Singletons/EmptyShowDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\EmptyShowDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/EmptyShowDeriving/' to rerun this test only.
    StandaloneDeriving:   FAIL (8.63s)
      1c1
      < Singletons/StandaloneDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\StandaloneDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/StandaloneDeriving/' to rerun this test only.
    T197:                 FAIL (3.64s)
      1c1
      < Singletons/T197.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T197.hs:(0,0)-(0,0): Splicing declarations

      Use -p '$0=="Testsuite.Singletons.T197"' to rerun this test only.
    T197b:                FAIL (2.35s)
      1c1
      < Singletons/T197b.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T197b.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T197b/' to rerun this test only.
    T200:                 FAIL (3.69s)
      1c1
      < Singletons/T200.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T200.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T200/' to rerun this test only.
    T204:                 FAIL (2.20s)
      1c1
      < Singletons/T204.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T204.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T204/' to rerun this test only.
    T206:                 OK (1.22s)
    T209:                 FAIL (2.27s)
      1c1
      < Singletons/T209.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T209.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T209/' to rerun this test only.
    T216:                 FAIL (2.02s)
      1c1
      < Singletons/T216.hs:0:0:: Splicing declarations
      ---
      > Singletons\T216.hs:0:0:: Splicing declarations

      Use -p '/T216/' to rerun this test only.
    T226:                 FAIL (1.74s)
      1c1
      < Singletons/T226.hs:0:0:: Splicing declarations
      ---
      > Singletons\T226.hs:0:0:: Splicing declarations

      Use -p '/T226/' to rerun this test only.
    T229:                 FAIL (3.28s)
      1c1
      < Singletons/T229.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T229.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T229/' to rerun this test only.
    T249:                 FAIL (2.09s)
      1c1
      < Singletons/T249.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T249.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T249/' to rerun this test only.
    OverloadedStrings:    FAIL (3.44s)
      1c1
      < Singletons/OverloadedStrings.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\OverloadedStrings.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/OverloadedStrings/' to rerun this test only.
    T271:                 FAIL (5.10s)
      1c1
      < Singletons/T271.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T271.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T271/' to rerun this test only.
    T287:                 FAIL (2.15s)
      1c1
      < Singletons/T287.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T287.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T287/' to rerun this test only.
    TypeRepTYPE:          OK (2.06s)
    T296:                 FAIL (2.12s)
      1c1
      < Singletons/T296.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T296.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T296/' to rerun this test only.
    T297:                 FAIL (2.05s)
      1c1
      < Singletons/T297.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T297.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T297/' to rerun this test only.
    T312:                 FAIL (2.33s)
      1c1
      < Singletons/T312.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T312.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T312/' to rerun this test only.
    T313:                 FAIL (2.27s)
      1c1
      < Singletons/T313.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T313.hs:(0,0)-(0,0): Splicing declarations
      63c63
      < Singletons/T313.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T313.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T313/' to rerun this test only.
    T316:                 FAIL (3.43s)
      1c1
      < Singletons/T316.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T316.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T316/' to rerun this test only.
    T322:                 FAIL (3.60s)
      1c1
      < Singletons/T322.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T322.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T322/' to rerun this test only.
    T326:                 FAIL (2.32s)
      1c1
      < Singletons/T326.hs:0:0:: Splicing declarations
      ---
      > Singletons\T326.hs:0:0:: Splicing declarations
      30c30
      < Singletons/T326.hs:0:0:: Splicing declarations
      ---
      > Singletons\T326.hs:0:0:: Splicing declarations

      Use -p '/T326/' to rerun this test only.
    NatSymbolReflexive:   OK (1.16s)
    T323:                 OK (1.13s)
    T332:                 FAIL (3.22s)
      1c1
      < Singletons/T332.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T332.hs:(0,0)-(0,0): Splicing declarations
      28c28
      < Singletons/T332.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T332.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T332/' to rerun this test only.
    T342:                 FAIL (2.17s)
      1c1
      < Singletons/T342.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T342.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T342/' to rerun this test only.
    FunctorLikeDeriving:  FAIL (12.17s)
      1c1
      < Singletons/FunctorLikeDeriving.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\FunctorLikeDeriving.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/FunctorLikeDeriving/' to rerun this test only.
    T353:                 FAIL (2.29s)
      1c1
      < Singletons/T353.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T353.hs:(0,0)-(0,0): Splicing declarations
      33c33
      < Singletons/T353.hs:0:0:: Splicing declarations
      ---
      > Singletons\T353.hs:0:0:: Splicing declarations
      73c73
      < Singletons/T353.hs:0:0:: Splicing declarations
      ---
      > Singletons\T353.hs:0:0:: Splicing declarations

      Use -p '/T353/' to rerun this test only.
    T358:                 FAIL (3.36s)
      1c1
      < Singletons/T358.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T358.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T358/' to rerun this test only.
    T367:                 FAIL (3.46s)
      1c1
      < Singletons/T367.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T367.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T367/' to rerun this test only.
    T371:                 FAIL (5.08s)
      1c1
      < Singletons/T371.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T371.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T371/' to rerun this test only.
    T376:                 FAIL (2.03s)
      1c1
      < Singletons/T376.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T376.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T376/' to rerun this test only.
    T378a:                FAIL (3.48s)
      1c1
      < Singletons/T378a.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T378a.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T378a/' to rerun this test only.
    T378b:                SKIP

      Use -p '/T378b/' to rerun this test only.
    T401:                 FAIL (1.50s)
      2c2
      < Singletons/T401.hs:0:0: error: Q monad failure
      ---
      > Singletons\T401.hs:0:0: error: Q monad failure
      7c7
      < Singletons/T401.hs:0:0: error:
      ---
      > Singletons\T401.hs:0:0: error:

      Use -p '$0=="Testsuite.Singletons.T401"' to rerun this test only.
    T401b:                FAIL (1.46s)
      2c2
      < Singletons/T401b.hs:0:0: error: Q monad failure
      ---
      > Singletons\T401b.hs:0:0: error: Q monad failure
      7c7
      < Singletons/T401b.hs:0:0: error:
      ---
      > Singletons\T401b.hs:0:0: error:

      Use -p '/T401b/' to rerun this test only.
    T402:                 FAIL (1.99s)
      1c1
      < Singletons/T402.hs:0:0:: Splicing declarations
      ---
      > Singletons\T402.hs:0:0:: Splicing declarations

      Use -p '/T402/' to rerun this test only.
    T410:                 FAIL (2.18s)
      1c1
      < Singletons/T410.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T410.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T410/' to rerun this test only.
    T412:                 FAIL (2.45s)
      1c1
      < Singletons/T412.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T412.hs:(0,0)-(0,0): Splicing declarations
      201c201
      < Singletons/T412.hs:0:0:: Splicing declarations
      ---
      > Singletons\T412.hs:0:0:: Splicing declarations

      Use -p '/T412/' to rerun this test only.
    T414:                 FAIL (1.97s)
      1c1
      < Singletons/T414.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T414.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T414/' to rerun this test only.
    T443:                 FAIL (2.18s)
      1c1
      < Singletons/T443.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T443.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T443/' to rerun this test only.
    T445:                 SKIP

      Use -p '/T445/' to rerun this test only.
    T450:                 FAIL (4.01s)
      1c1
      < Singletons/T450.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T450.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T450/' to rerun this test only.
    T453:                 FAIL (1.95s)
      1c1
      < Singletons/T453.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T453.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T453/' to rerun this test only.
    NegativeLiterals:     FAIL (2.02s)
      1c1
      < Singletons/NegativeLiterals.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\NegativeLiterals.hs:(0,0)-(0,0): Splicing declarations
      17c17
      < Singletons/NegativeLiterals.hs:0:0: warning: [-Woverflowed-literals]
      ---
      > Singletons\NegativeLiterals.hs:0:0: warning: [-Woverflowed-literals]

      Use -p '/NegativeLiterals/' to rerun this test only.
    T470:                 FAIL (3.17s)
      1c1
      < Singletons/T470.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T470.hs:(0,0)-(0,0): Splicing declarations
      92c92
      < Singletons/T470.hs:0:0: warning:
      ---
      > Singletons\T470.hs:0:0: warning:

      Use -p '/T470/' to rerun this test only.
    T480:                 FAIL (1.40s)
      2c2
      < Singletons/T480.hs:0:0: error: Q monad failure
      ---
      > Singletons\T480.hs:0:0: error: Q monad failure
      7c7
      < Singletons/T480.hs:0:0: error:
      ---
      > Singletons\T480.hs:0:0: error:

      Use -p '/T480/' to rerun this test only.
    T487:                 FAIL (3.53s)
      1c1
      < Singletons/T487.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T487.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T487/' to rerun this test only.
    T489:                 FAIL (3.41s)
      1c1
      < Singletons/T489.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\T489.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T489/' to rerun this test only.
    T492:                 OK (1.63s)
    Natural:              FAIL (3.31s)
      1c1
      < Singletons/Natural.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Singletons\Natural.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Natural/' to rerun this test only.
  Promote
    Constructors:         FAIL (2.08s)
      1c1
      < Promote/Constructors.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Promote\Constructors.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Constructors/' to rerun this test only.
    GenDefunSymbols:      FAIL (2.13s)
      1c1
      < Promote/GenDefunSymbols.hs:0:0:: Splicing declarations
      ---
      > Promote\GenDefunSymbols.hs:0:0:: Splicing declarations

      Use -p '/GenDefunSymbols/' to rerun this test only.
    Newtypes:             SKIP

      Use -p '/Newtypes/' to rerun this test only.
    Pragmas:              FAIL (2.10s)
      1c1
      < Promote/Pragmas.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Promote\Pragmas.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Pragmas/' to rerun this test only.
    Prelude:              FAIL (3.21s)
      1c1
      < Promote/Prelude.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Promote\Prelude.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/Prelude/' to rerun this test only.
    T180:                 FAIL (2.76s)
      1c1
      < Promote/T180.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Promote\T180.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T180/' to rerun this test only.
    T361:                 FAIL (1.69s)
      1c1
      < Promote/T361.hs:0:0:: Splicing declarations
      ---
      > Promote\T361.hs:0:0:: Splicing declarations
      7c7
      < Promote/T361.hs:(0,0)-(0,0): Splicing declarations
      ---
      > Promote\T361.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/T361/' to rerun this test only.
  Database client
    Database:             FAIL (6.87s)
      1c1
      < GradingClient/Database.hs:(0,0)-(0,0): Splicing declarations
      ---
      > GradingClient\Database.hs:(0,0)-(0,0): Splicing declarations
      164c164
      < GradingClient/Database.hs:(0,0)-(0,0): Splicing declarations
      ---
      > GradingClient\Database.hs:(0,0)-(0,0): Splicing declarations
      3471c3471
      < GradingClient/Database.hs:0:0:: Splicing declarations
      ---
      > GradingClient\Database.hs:0:0:: Splicing declarations
      3473c3473
      < GradingClient/Database.hs:(0,0)-(0,0): Splicing expression
      ---
      > GradingClient\Database.hs:(0,0)-(0,0): Splicing expression

      Use -p '/Database client.Database/' to rerun this test only.
    Main:                 SKIP

      Use -p '/Main/' to rerun this test only.
  InsertionSort
    InsertionSortImp:     FAIL (2.70s)
      1c1
      < InsertionSort/InsertionSortImp.hs:(0,0)-(0,0): Splicing declarations
      ---
      > InsertionSort\InsertionSortImp.hs:(0,0)-(0,0): Splicing declarations
      39c39
      < InsertionSort/InsertionSortImp.hs:(0,0)-(0,0): Splicing declarations
      ---
      > InsertionSort\InsertionSortImp.hs:(0,0)-(0,0): Splicing declarations

      Use -p '/InsertionSortImp/' to rerun this test only.

114 out of 121 tests failed (34.63s)

Test suite singletons-base-test-suite: FAIL
```
</details>

This is because GHC on Windows will display path separators as `Foo\Bar.hs` rather than `Foo/Bar.hs`. To address this, we should find a way to normalize path separators in the `turtle` script that we use in the test suite. We could adopt inspiration from how GHC's test suite handles this [here](https://gitlab.haskell.org/ghc/ghc/-/blob/master/testsuite/driver/testlib.py#L2299-2300).

--- a/singletons-base/tests/SingletonsBaseTestSuiteUtils.hs
+++ b/singletons-base/tests/SingletonsBaseTestSuiteUtils.hs
@@ -23,6 +23,7 @@ import System.Process        ( CreateProcess(..), StdStream(..)
                              , callCommand                         )
 import Test.Tasty            ( TestTree, testGroup                 )
 import Test.Tasty.Golden     ( goldenVsFileDiff                    )
+import qualified Data.Text as Text
 import qualified Turtle
 
 -- Some infractructure for handling external process errors
@@ -140,6 +141,38 @@ are turned into:
 
 This allows inserting comments into test files without the need to modify the
 golden file to adjust line numbers.
+
+Note [Normalizing Windows path separators]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+File paths are printed differently by GHC depending on which operating system
+one uses:
+
+* On Unix-like operating systems: Foo/Bar.hs
+* On Windows:                     Foo\Bar.hs
+
+This is annoying for golden testing, since it means that checking in output
+that prints file paths will only work on some operating systems and not others.
+To circumvent this problem, we normalize Windows-style path separators to
+Unix-like ones.
+
+One way to do this is to replace all occurrences of the '\' character with '/'.
+This is a step too far, however, since this will normalize things like
+(\x -> x) to (/x -> x). A reasonable middle ground is to require that the
+characters before and after the '\' character are alphanumeric before
+normalizing it. This just so happens to be the case for every path that is
+checked into the singletons-base test suite. This approach isn't perfect, since
+one could check in a golden file that prints a path with a non-alphanumeric
+character. This seems very unlikely to happen, however, so we will only worry
+about that issue should it ever arise in practice.
+
+Moreover, this approach would normalize expressions like (id\x -> x)
+to (id/x -> x), which could hypothetically happen with BlockArguments. However,
+the vast majority of the expressions that we check into golden files arise from
+-ddump-splices output, which puts surrounding whitespace after the backslashes
+in lambda expressions. As a result, the expression above would be
+pretty-printed as (id \ x -> x), which avoids the issue entirely. Again, we
+will choose not to worry about this corner case unless it becomes an issue in
+practice.
 -}
 
 normalizeOutput :: FilePath -> IO ()
@@ -156,6 +189,10 @@ normalizeOutput file = Turtle.inplace pat (fromString file)
       -- (e.g., turn `singletons-2.4.1:Sing` into `Sing`) to make the output
       -- more stable.
       , "" <$ "singletons-" <* verNum <* ":"
+      , do x <- Turtle.alphaNum
+           _ <- Turtle.char '\\'
+           y <- Turtle.alphaNum
+           pure $ Text.pack [x, '/', y]
       ]
     verNum = d `Turtle.sepBy` Turtle.char '.'
     numPair = () <$ "(" <* d <* "," <* d <* ")"

